source/local/local.go

```diff
@@ -321,6 +324,23 @@ func getFeaturesFromFiles() (map[string]string, error) {
 
 	for _, file := range files {
 		fileName := file.Name()
+
+		fileInfo, err := file.Info()
+		if err != nil {
+			klog.ErrorS(err, "failed to get file info", "fileName", fileName)
+			continue
+		}
+
+		fileSize := fileInfo.Size()
+		if fileSize > MaxFeatureFileSize {
+			klog.ErrorS(
+				fmt.Errorf("invalid file size: %d bytes > %d bytes", fileSize, MaxFeatureFileSize),
+				"skipping file with invalid size",
+				"fileName", fileName, "fileSize", fileSize,
```

Comment:
nit: suggesting a small change, e.g. "too big" instead of "invalid" 

```suggestion
				fmt.Errorf("file size limit exceeded: %d bytes > %d bytes", fileSize, MaxFeatureFileSize),
				"skipping too big feature file",
				"fileName", fileName,
```